torch/_subclasses/fake_tensor.py

```diff
@@ -422,11 +422,53 @@ def nyi(fake_mode, func, *args, **kwargs):
     assert func not in _device_not_kwarg_ops, f"NYI: {func}"
 
 
-# Meta tensors give you the ability to run PyTorch code without having to
-# actually do computation through tensors allocated on a `meta` device.
-# Because the device is `meta`, meta tensors do not model device propagation.
-# FakeTensor extends MetaTensors to also carry an additional `fake_device`
-# which tracks devices that would have been used.
+@register_op_impl(
+    lambda func: func in (aten.convolution.default, aten.convolution_backward.default)
```

Comment:
So the idea is that the conv meta impl in `_meta_registrations.py` is mostly correct, but gives the wrong memory format - so we manually call it and then patch up the memory format here in `FakeTensorMode`?

Is there any reason not to just patch the fancier memory format logic directly into the existing meta function? (If the answer is just that's more effort and work for later then totally fair - just want to confirm my understanding).

Reply:
Most of the complexity of the logic comes in cpu, so it's not that high pri at the moment, but it was a good way to unblock turning on fake tensors for inductor shape prop.

We will eventually need to patch the fancier memory format logic yea - but there might be a way to do something similar as we're doing here even for symbolic shapes

Reply:
This logic should be moved into `_meta_registrations.py`

Reply:
Yea, I did a best effort fix [here](https://github.com/pytorch/torchdynamo/pull/1639). There is a [lot](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L193) [lot](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L239-L273) [lot](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Convolution.cpp#L313-L325) of stuff to copy over.